lib/trino-hive-formats/src/main/java/io/trino/hive/formats/line/text/TextLineReaderFactory.java

```diff
@@ -77,11 +77,12 @@ public LineReader createLineReader(
             LineReader lineReader = new TextLineReader(inputStream, fileBufferSize, start, length);
 
             //  Only skip header rows when the split is at the beginning of the file
-            if (headerCount > 0) {
+            if (headerCount > 0 && start == 0) {
```

Comment:
I think we need a similar check here. We can't support splitting for files with multiple headers, since the first header could be larger than the first split, but the subsequent splits wouldn't know to skip the additional headers.